### ISSUE TITLE
[Request] Configuring LSP server using project module

### ISSUE THREAD

**Issue by FijiHasGithub** [lite-xl/lite-xl-lsp#89]
Currently, if you want to configure something at runtime (eg; change the command for a specific project) you need to do something like this in your `.lite_project.lua`
```lua
local lsp = require "plugins.lsp"
lsp.servers.server_name.option = value
```

Real world example of adding another path to Jedi's `init_options`:
```lua
local lsp = require "plugins.lsp"
lsp.servers.jedi.init_options.workspace.extraPaths[2] = "foo"
```

---

I think it would be easier if you could configure things using `lspconfig`, something like this:
```lua
local lspconfig = require "plugins.lsp.config"
lspconfig.jedi.init_options.workspace.extraPaths[2] = "foo"
```
instead of digging through the code to find how to configure settings

**Comment by Guldoman** [comment]
What do you think about adding a function to retrieve a server table by name?
Mostly because `lspconfig` only contains the "preconfigured" servers, and it wouldn't contain anything added directly via `lsp.add_server`.

So something like
```lua
local sumneko = lsp.get_server_by_name("lua-language-server")
sumneko.settings.Lua.signatureHelp.enable = false
```

Or even something like
```lua
lsp.update_server_config("lua-language-server", {
  settings = {
    Lua = {
      signatureHelp = { enable = false }
    }
  }
})
```

**Comment by FijiHasGithub** [comment]
yeah that seems much better, I proposed `lspconfig` because it looked like the most obvious place to configure LSP servers

**Comment by Guldoman** [comment]
Until the end of the month I won't be able to implement this as I don't have much time.
Until then, anyone that wants to give this a try, I'd be glad to review and merge their proposal!
